src/packages/auth/src/authentication/resolvers/api-key/resolver.ts

```diff
@@ -0,0 +1,89 @@
+import {
+	BackendProvider,
+	CreateOrUpdateHookParams,
+	GraphqlEntityType,
+	HookParams,
+	HookRegister,
+	Resolver,
+	WithId,
+	hookManagerMap,
+} from '@exogee/graphweaver';
+import { AuthenticationError, ValidationError } from 'apollo-server-errors';
+
+import { ApiKeyStorage } from '../../entities';
+import {
+	ApiKeyInputArgs,
+	ApiKeyCreateOrUpdateInputArgs,
+	createApiKeyBaseResolver,
+} from './base-resolver';
+import { hashPassword } from '../../../utils/argon2id';
+
+export const createApiKeyResolver = <G extends WithId, D extends ApiKeyStorage>(
+	gqlEntityType: GraphqlEntityType<G, D>,
+	provider: BackendProvider<D, G>
+) => {
+	@Resolver()
+	class ApiKeyAuthResolver extends createApiKeyBaseResolver(gqlEntityType, provider) {
+		provider = provider;
+
+		public async runAfterHooks<
+			H extends HookParams<ApiKeyInputArgs | ApiKeyCreateOrUpdateInputArgs>
+		>(hookRegister: HookRegister, hookParams: H, entities: (D | null)[]): Promise<(D | null)[]> {
+			const hookManager = hookManagerMap.get('ApiKey');
+			const { entities: hookEntities = [] } = hookManager
+				? await hookManager.runHooks(hookRegister, {
+						...hookParams,
+						entities,
+				  })
+				: { entities };
+
+			return hookEntities;
+		}
+
+		async create(params: CreateOrUpdateHookParams<ApiKeyInputArgs>): Promise<D> {
+			const [item] = params.args.items;
+			if (!item) throw new ValidationError('No data specified cannot continue.');
+
+			if (!item.key) throw new ValidationError('Create unsuccessful: Key not defined.');
+
+			if (!item.secret) throw new ValidationError('Create unsuccessful: Secret not defined.');
+
+			const secretHash = await hashPassword(item.secret);
+			const apiKey = await this.provider.createOne({
+				key: item.key,
+				secret: secretHash,
+				...(Object(item).hasOwnProperty('revoked') ? { revoked: item.revoked } : {}),
```

Comment:
Is this saying "Hey you're creating a key that's revoked so I'm setting it to revoked?